dask/array/core.py

```diff
@@ -5771,23 +5771,24 @@ def _subset_to_indexed_axes(iterable):
                 key = sorted_keys[start]
                 outblock, *input_blocks = np.unravel_index(key, ravel_shape)
                 inblock = [_[slicer] for _ in sorted_inblock_idxs]
-                dsk[keyname(name, i, okey)] = (
-                    _vindex_transpose,
-                    (
-                        _vindex_slice,
-                        (x.name,) + interleave_none(okey, input_blocks),
-                        interleave_none(full_slices, inblock),
-                    ),
+                k = keyname(name, i, okey)
+                dsk[k] = Task(
+                    k,
+                    _vindex_slice_and_transpose,
+                    TaskRef((x.name,) + interleave_none(okey, input_blocks)),
+                    interleave_none(full_slices, inblock),
                     axis,
                 )
-                merge_inputs[outblock].append(keyname(name, i, okey))
+                merge_inputs[outblock].append(TaskRef(keyname(name, i, okey)))
                 merge_indexer[outblock].append(sorted_outblock_idx[slicer])
 
             for i in merge_inputs.keys():
-                dsk[keyname(vindex_merge_name, i, okey)] = (
+                k = keyname(vindex_merge_name, i, okey)
+                dsk[k] = Task(
+                    k,
                     _vindex_merge,
                     merge_indexer[i],
-                    merge_inputs[i],
+                    List(merge_inputs[i]),
```

Comment:
nit: Could we get rid of this in light of https://github.com/dask/distributed/issues/8958?

Reply:
had a brief offline discussion and decided to keep this for now until we make a general effort to get rid of these patterns